applicationset/generators/plugin.go

```diff
@@ -86,12 +88,19 @@ func (g *PluginGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.App
 	return res, nil
 }
 
-func (g *PluginGenerator) getPluginFromGenerator(ctx context.Context, appSetName string, generatorConfig *argoprojiov1alpha1.PluginGenerator) (*plugin.Service, error) {
-	cm, err := g.getConfigMap(ctx, generatorConfig.ConfigMapRef.Name)
+func (g *PluginGenerator) getPluginConfigNamespaceOrDefaultToAppSetNamespace(plugin *argoprojiov1alpha1.PluginGenerator, applicationSetInfo *argoprojiov1alpha1.ApplicationSet) string {
+	if plugin.ConfigMapRef.Namespace != "" {
+		return plugin.ConfigMapRef.Namespace
+	}
+	return applicationSetInfo.Namespace
+}
+
+func (g *PluginGenerator) getPluginFromGenerator(ctx context.Context, appSetName, namespace string, generatorConfig *argoprojiov1alpha1.PluginGenerator) (*plugin.Service, error) {
+	cm, err := g.getConfigMap(ctx, generatorConfig.ConfigMapRef.Name, namespace)
 	if err != nil {
 		return nil, fmt.Errorf("error fetching ConfigMap: %w", err)
 	}
-	token, err := g.getToken(ctx, cm["token"])
+	token, err := g.getToken(ctx, cm["token"], namespace)
```

Comment:
We need to consider whether we want the plugin ConfigMaps to be able to reference Secrets from only the `argocd` namespace, only the AppSet namespace, or both. If both, we need to define the ConfigMap syntax for specifying which to reference.